cornflow/schemas/schema_manager.py

```diff
@@ -0,0 +1,368 @@
+            )
+        elif "$ref" in content:
+            schema_name = self._get_ref((None, content))
+        else:
+            self.schema_dict.update(
+                {schema_name: [self._get_element_dict(i, self._get_required(content)) for i in content.items()]})
+        return schema_name
+       
+    def _get_field_dict(self, item, required_list=None):
+        """
+        Transform a "normal" item from the jsonschema in a dict for the schema_dict and return it.
+        This is used for items that will directly translate into fields.
+        
+        :param item: The jsonschema item in format (key, value)
+        :param required_list: a list of the fields required in the parent object.
+        
+        :return: the schema_dict for this item
+        """
+        
+        d = dict(
+            name=item[0],
+            type=self._get_type(item[1]["type"]),
+            required=(item[0] in required_list),
+            allow_none=("null" in item[1]["type"]),
+            many=False)
+        return d
+    
+    def _get_ref(self, item):
+        """
+        Get the name of the schema for a jsonschema reference.
+        jsonschema definitions are parsed first and corresponign schema are created so a schema should exist
+         corresponding to the reference.
+        
+        :param item: The jsonschema item in format (key, value)
+        The value should be in the following format: {"$ref": "#/definitions/object_name"}
+        
+        :return The schema name (_get_schema_name(object_name))
+        """
+        content = item[1]
+        ref = re.search("definitions/(.+)", content["$ref"]).group(1)
+        return self._get_schema_name(ref)
+    
+    def _get_type(self, json_type):
+        """
+        Translate the type between jsonschema and schema_dict.
+        
+        :param json_type: the type in jsonschema
+        
+        :return: the type in schema_dict.
+        """
+        if type(json_type) is list:
+            if "null" in json_type:
+                json_type = [i for i in json_type if i != "null"]
+            if len(json_type) > 1:
+                raise Warning("Warning: more than one type given")
+            return self.types[json_type[0]]
+        else:
+            return self.types[json_type]
+    
+    @staticmethod
+    def _get_schema_name(name):
+        """
+        Transform an element name into a schema name in order to create a schema corrsponing to an object or array.
+        
+        :param name: The name of the object or array.
+        
+        :return: the corresponign schema name.
+        """
+        return name.capitalize() + "Schema"
+    
+    @staticmethod
+    def _get_required(content):
+        """
+        Get the list of required name of it exist.
+        
+        :content: the dict which should have a "required" key.value
+        
+        :return: The required list or empty list.
+        
+        """
+        if "required" in content:
```

Comment:
you can also do something like: `return content.get('required', [])`

Reply:
done